spec/datadog/tracing/contrib/sinatra/multi_app_spec.rb

```diff
@@ -74,7 +74,7 @@
           spans.each do |span|
             if span.name == Datadog::Tracing::Contrib::Rack::Ext::SPAN_REQUEST
               expect(span.resource).to eq('GET /endpoint')
-              expect(span.get_tag(Datadog::Tracing::Metadata::Ext::HTTP::TAG_URL)).to eq('/endpoint')
+              expect(span.get_tag(Datadog::Tracing::Metadata::Ext::HTTP::TAG_URL)).to eq('/one/endpoint')
```

Comment:
Could you explain why these Sinatra tests have changed?

Reply:
These tags actually come from Rack, not Sinatra. I believe they're here because at some point Sinatra may have been backfilling this value by itself. In any case they act as an integration test checking for side effects.

Sadly this test would previously pass but not represent what happens when real-world `REQUEST_URI` is set, indeed `Rack::Test` does not set `REQUEST_URI`.

The change here is mainly due to a bug that this PR addresses, that previously made the fallback to `PATH_INFO` be inconsistent with `REQUEST_URI` (i.e it did not include `SCRIPT_NAME` nor `QUERY_STRING`), which created all kinds of special cases from the lack of query string to adding `base_url tag + url tag` not resulting in the same value depending on whether `REQUEST_URI` or `PATH_INFO` was used, not the least in the latter case `base_url tag + url tag` is entirely nonsensical for virtualhosted or nested apps since it misses the intermediate `SCRIPT_NAME`.